helix-core/src/test/java/org/apache/helix/controller/rebalancer/waged/TestWagedRebalancer.java

```diff
@@ -440,15 +444,15 @@ public void testRebalanceOnChanges() throws IOException, HelixRebalanceException
     // calculate the assignment for both resources.
     newIdealStates =
         rebalancer.computeNewIdealStates(clusterData, resourceMap, new CurrentStateOutput());
-    algorithmResult = _algorithm.getRebalanceResult();
+    newAlgorithmResult = _algorithm.getRebalanceResult();
     // Verify that both resource has been included in the calculation.
-    validateRebalanceResult(resourceMap, newIdealStates, algorithmResult);
-    // Both assignment state should be empty since no cluster topology change.
+    validateRebalanceResult(resourceMap, newIdealStates, newAlgorithmResult);
+    // There is nothing changes the baseline.
     baseline = _metadataStore.getBaseline();
-    Assert.assertEquals(baseline, Collections.emptyMap());
-    // The best possible assignment should be present.
+    Assert.assertEquals(baseline, algorithmResult);
+    // The best possible assignment has been updated as long as computeNewIdealStates() is called.
```

Comment:
The BestPossible assignment should have been updated since computeNewIdealStates() should have been called.